src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -430,6 +436,22 @@ internal static int OptimalMaxSpinWaitsPerSpinIteration
             get;
         }
 
+        private static class DirectOnThreadLocalData
+        {
+            // Special Thread Static variable which is always allocated at the address of the Thread variable in the ThreadLocalData of the current thread
+            [ThreadStatic]
+            public static IntPtr pNativeThread;
+        }
+        /// <summary>
+        /// Get the ThreadStaticBase used for this threads TLS data
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [System.Runtime.BypassReadyToRunAttribute]
```

Comment:
Can we get a comment in a `<remarks>` block that describes the why this is correct?